perun-core/src/main/java/cz/metacentrum/perun/core/impl/AuthzResolverImpl.java

```diff
@@ -902,6 +917,120 @@ public void unsetRole(PerunSession sess, Map<String, Integer> mappingOfValues, S
 		}
 	}
 
+	@Override
+	public List<Vo> getVosWhereUserIsInRoles(User user, List<String> roles) {
+		MapSqlParameterSource parameters = prepareParametersToGetObjectsByuserRoles(user, roles);
+
+		try {
+			return namedParameterJdbcTemplate.query("select " + voMappingSelectQuery + " from authz join vos on authz.vo_id=vos.id " +
```

Comment:
You should prevent duplicates in the list (use set or something like that). Should be used for all these methods.

Reply:
All Impl methods return Set of objects now. It is converted to ArrayList in the Bl layer.